utils/httputil/client.go

```diff
@@ -230,10 +234,34 @@ func (c *Client) request(
 			continue
 		}
 
-		if status = r.GetStatus(); status == StatusTooManyRequests || status >= 500 {
+		status = r.GetStatus()
+
+		if status == StatusNirnTimeout || status >= 500 {
```

Comment:
Rather than adding a `StatusNirnTimeout`, I think we should just write `status == http.StatusRequestTimeout`.